### PR TITLE
[BT-817] Add nullable model_name to classifier, add auth to load_model calls in train classifier, documentation

### PR DESCRIPTION
**Pull request by niole**
- adds  base_model_name column to classifier and migration with default value
- updates classifier schema to contain base_model_name with default value
- passes model name to train classifier logic, defaults to a default model if not specified in DB
- adds HF auth to bumblebee calls

### CODE REVIEW

File: lib/excision/excisions/classifier.ex [hunk 1]
```diff
@@ -2,8 +2,13 @@ defmodule Excision.Excisions.Classifier do
   use Ecto.Schema
   import Ecto.Changeset
 
+  def default_model_name do
+    "distilbert/distilbert-base-uncased"
```
Comment (niole) [inline lib/excision/excisions/classifier.ex:6]: TODO use smaller llm

File: lib/excision/excisions.ex [hunk 2]
```diff
@@ -357,8 +357,24 @@ defmodule Excision.Excisions do
 
   """
   def create_classifier(attrs \\ %{}) do
```
Comment (feynmanliang) [inline lib/excision/excisions.ex:359]: Can we use https://hexdocs.pm/ecto/Ecto.Schema.html#field/3-options `:default`

File: lib/excision/workers/train_classifier.ex [hunk 3]
```diff
@@ -113,15 +110,18 @@ defmodule Excision.Workers.TrainClassifier do
   end
 
   defp load_model_and_tokenizer(model_name, num_labels, sequence_length) do
+    # TODO surface errors to user better
```
Comment (feynmanliang) [inline lib/excision/workers/train_classifier.ex:113]: (1) Can we use `Application.get_env(...)` and read from environment in `runtime.exs`?
(2) Does this still work if `HUGGING_FACE_API_KEY` does not exist?

File: lib/excision/excisions/classifier.ex [hunk 4]
```diff
@@ -38,6 +43,7 @@ defmodule Excision.Excisions.Classifier do
     classifier
     |> cast(attrs, [
       :name,
+      :model_name,
```
Comment (feynmanliang) [inline lib/excision/excisions/classifier.ex:46]: Can we call this `base_model_name`?

File: priv/repo/migrations/20240929163620_add_classifier_model_name.exs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+defmodule Excision.Repo.Migrations.AddClassifierModelName do
+  use Ecto.Migration
+
+  def change do
+    # add nullable column model_name
+    alter table(:classifiers) do
+      add :model_name, :string, null: true
```
Comment (feynmanliang) [inline priv/repo/migrations/20240929163620_add_classifier_model_name.exs:7]: can we call this base_model_name? and comment that `null: true` means distilbert, and that it is not safe to change the default until we migrate these legacy records' `model_name`s?

**Review by feynmanliang — DISMISSED** [review]

File: lib/excision/workers/train_classifier.ex (hunk 3 above)
Comment (niole) [reply lib/excision/workers/train_classifier.ex:113]: done, yes works if key doesn't exist

File: priv/repo/migrations/20240929163620_add_classifier_model_name.exs (hunk 5 above)
Comment (niole) [reply priv/repo/migrations/20240929163620_add_classifier_model_name.exs]: updated

File: priv/repo/migrations/20240929163620_add_classifier_model_name.exs [hunk 6]
```diff
@@ -0,0 +1,12 @@
+defmodule Excision.Repo.Migrations.AddClassifierModelName do
+  use Ecto.Migration
+
+  def change do
+    # add nullable column base_model_name
+    # if base_model_name not provided, distilbert/distilbert-base-uncased is set
+    # app may not work as expected until you run the migration
+    alter table(:classifiers) do
+      add :base_model_name, :string, null: true, default: "distilbert/distilbert-base-uncased"
```
Comment (niole) [inline priv/repo/migrations/20240929163620_add_classifier_model_name.exs:9]: doesn't need to be nullable

**Pull request merged**